traitsui/tests/editors/test_button_editor.py

```diff
@@ -85,3 +89,39 @@ def test_simple_button_editor(self):
     @skip_if_null
     def test_custom_button_editor(self):
         self.check_button_text_update(custom_view)
+
+
+@skip_if_not_qt4
+class TestButtonEditorValuesTrait(unittest.TestCase):
+    """ The values_trait is only supported by Qt.
+
+    See discussion enthought/traitsui#879
+    """
+
+    def get_view(self, style):
+        return View(
+            Item(
+                "play_button",
+                editor=ButtonEditor(values_trait="values"),
+                style=style,
+            ),
+        )
+
+    def check_editor_values_trait_init_and_dispose(self, style):
+        # Smoke test to check init and dispose when values_trait is used.
+        instance = ButtonTextEdit(values=["Item1", "Item2"])
+        view = self.get_view(style=style)
+        with store_exceptions_on_all_threads():
+            with create_ui(instance, dict(view=view)):
```

Comment:
Is this okay without `gui.process_events()`? There are quite a few qt methods called in the code.

Reply:
I am about to open a PR to add GUI event queue processing prior to all disposal of UIs and convert all tests to do that. For this test, nothing external to the editor's code causes new events in between `init` and `dispose`, so this is okay, but we should do this flush more generally.

I also noticed that the Qt implementation for `GUI.process_events` does not actually do what it says in the tin if one reads it without thinking/reading harder about how Qt works.  That is going to be addressed in that separate PR mentioned above.

Reply:
Could I NOT add this `gui.process_events()` here? For the following reasons:
(1) `create_ui` should do that clean up, not here
(2) If I add it here as a "just-in-case" measure, it would be confusing to others whether it is generally needed for all the tests that just calls `create_ui` for opening and closing the UI. 
(3) There are no race conditions for the setup and teardown of this simple button editor as far as I can see.

Confusion is one of the drawbacks of this kind of defensive programming. For (3), I think it is worth avoiding that confusion.